src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -0,0 +1,106 @@
+package assisted_installer_controller
+
+import (
+	"context"
+	"time"
+
+	"github.com/eranco74/assisted-installer/src/inventory_client"
+	"github.com/eranco74/assisted-installer/src/k8s_client"
+	"github.com/eranco74/assisted-installer/src/ops"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	done = "Done"
+)
+
+// assiste
+
+type ControllerConfig struct {
+	ClusterID string `envconfig:"CLUSTER_ID" required:"true" `
+	Host      string `envconfig:"INVENTORY_HOST" required:"true"`
+	Port      int    `envconfig:"INVENTORY_PORT" required:"true"`
+}
+
+type Controller interface {
+	WaitAndUpdateNodesStatus()
+}
+
+type controller struct {
+	ControllerConfig
+	log                    *logrus.Logger
+	ops                    ops.Ops
+	ic                     inventory_client.InventoryClient
+	kc                     k8s_client.K8SClient
+	assistedInstallerNodes []string
+}
+
+func NewController(log *logrus.Logger, cfg ControllerConfig, ops ops.Ops, ic inventory_client.InventoryClient, kc k8s_client.K8SClient) *controller {
+	return &controller{
+		log:              log,
+		ControllerConfig: cfg,
+		ops:              ops,
+		ic:               ic,
+		kc:               kc,
+	}
+}
+
+func (c *controller) WaitAndUpdateNodesStatus() {
+	c.log.Infof("Waiting till all nodes will join and update status to assisted installer")
+
+	c.getInventoryNodes()
+
+	apiContext, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	nodeNameAndUUID := make(map[string]string)
+	wait.Until(func() {
+		nodes, err := c.getNodes()
+		if err != nil {
+			return
+		}
+		for _, node := range nodes.Items {
+			if _, ok := nodeNameAndUUID[node.Name]; ok {
+				continue
+			}
+			c.log.Infof("Found new joined node %s with id %s, updating its status to %s", node.Name, node.Status.NodeInfo.SystemUUID, done)
+			if err := c.ic.UpdateHostStatus(done, node.Status.NodeInfo.SystemUUID); err != nil {
+				c.log.Errorf("Failed to update node installation status, %s", err)
+				return
+			}
+
+			nodeNameAndUUID[node.Name] = node.Status.NodeInfo.SystemUUID
+		}
+		c.log.Infof("Found %d nodes: %+v", len(nodes.Items), nodeNameAndUUID)
+		if len(nodeNameAndUUID) >= len(c.assistedInstallerNodes) {
+			c.log.Infof("WaitAndUpdateNodesStatus - Done")
+			cancel()
+		}
+	}, 30*time.Second, apiContext.Done())
+}
+
+func (c *controller) getInventoryNodes() {
+	c.log.Infof("Getting list of inventory nodes")
+	apiContext, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	wait.Until(func() {
+		var err error
```

Comment:
Why do you need this?

Reply:
cause i am using c.assistedInstallerNodes and i want to print error too.
i could not to c.assistedInstallerNodes, err := c.ic.GetHostsIds()
thats why i need to create error before

Reply:
Consider returning the inventoryNodes from this method.
You named it `getInventoryNodes` Not `setInventoryNodes`
It should also solve this issue.

Reply:
explained why

Reply:
will change name to setInventoryNodes